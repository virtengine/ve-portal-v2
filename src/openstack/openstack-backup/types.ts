import { Resource, InternalIP } from '@waldur/resource/types';

import { SecurityGroupRule } from '../types';

interface NestedSecurityGroup {
  url: string;
  name: string;
  description: string;
  rules: SecurityGroupRule[];
}

export interface OpenStackBackup extends Resource {
  instance_security_groups: NestedSecurityGroup[];
  instance_internal_ips_set: InternalIP[];
}
