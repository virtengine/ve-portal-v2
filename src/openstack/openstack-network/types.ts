import { Resource } from '@waldur/resource/types';

export interface Network extends Resource {
  type: string;
  segmentation_id?: number;
  is_external: boolean;
}
