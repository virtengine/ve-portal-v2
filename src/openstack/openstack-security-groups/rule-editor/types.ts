export interface Rule {
  ethertype: string;
  direction: string;
  protocol: string;
  from_port?: number;
  to_port?: number;
  cidr?: string;
  remote_group?: string;
  description?: string;
}

export interface FormData {
  rules: Rule[];
}
