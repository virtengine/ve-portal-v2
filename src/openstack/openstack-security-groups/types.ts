import { SecurityGroupRule } from '../types';

export interface SecurityGroup {
  url: string;
  uuid: string;
  name: string;
  settings: string;
  description: string;
  rules: SecurityGroupRule[];
  resource_type?: string;
  tenant?: string;
}

export interface SecurityGroupOption extends SecurityGroup {
  clearableValue?: boolean;
}
