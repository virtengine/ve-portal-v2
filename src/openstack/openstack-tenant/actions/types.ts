import { OpenStackTenant } from '../types';

export interface TenantActionProps {
  resource: OpenStackTenant;
}
