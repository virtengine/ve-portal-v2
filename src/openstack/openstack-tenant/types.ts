import { Resource } from '@waldur/resource/types';

export interface OpenStackTenant extends Resource {
  extra_configuration?: {
    package_uuid?: string;
  };
  internal_network_id?: string;
  external_network_id?: string;
  access_url?: string;
  user_username?: string;
  user_password?: string;
}
