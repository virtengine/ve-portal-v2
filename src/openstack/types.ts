export interface Quota {
  name: string;
  limit: number;
  usage: number;
  limitType?: string;
  required?: number;
}

export interface AvailabilityZone {
  url: string;
  name: string;
}
