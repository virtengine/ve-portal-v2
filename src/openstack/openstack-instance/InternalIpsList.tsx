import { FunctionComponent } from 'react';

import { getById } from '@waldur/core/api';
import { VirtualMachine, InternalIP } from '@waldur/resource/types';
import { Table, connectTable } from '@waldur/table';

import { UpdateInternalIpsAction } from './actions/update-internal-ips/UpdateInternalIpsSetAction';
import { SetAllowedAddressPairsButton } from './SetAllowedAddressPairsButton';

const TableComponent: FunctionComponent<any> = (props) => {
  const { translate } = props;
  return (
    <Table<InternalIP>
      {...props}
      columns={[
        {
          title: translate('IP address'),
          render: ({ row }) =>
            row.fixed_ips.map((fip) => fip.ip_address).join(', ') || 'N/A',
        },
        {
          title: translate('MAC address'),
          render: ({ row }) => row.mac_address,
        },
        {
          title: translate('Subnet name'),
          render: ({ row }) => row.subnet_name,
        },
        {
          title: translate('Subnet CIDR'),
          render: ({ row }) => row.subnet_cidr,
        },
        {
          title: translate('Actions'),
          render: ({ row }) => (
            <SetAllowedAddressPairsButton
              instance={props.resource}
              internalIp={row}
            />
          ),
        },
      ]}
      verboseName={translate('internal IPs')}
      actions={<UpdateInternalIpsAction resource={props.resource} />}
    />
  );
};

const getInternalIps = (request) =>
  getById<VirtualMachine>(
    '/openstacktenant-instances/',
    request.filter.uuid,
  ).then((vm) => ({
    rows: vm.internal_ips_set,
    resultCount: vm.internal_ips_set.length,
  }));

const TableOptions = {
  table: 'openstack-internal-ips',
  fetchData: getInternalIps,
  mapPropsToFilter: (props) => ({
    uuid: props.resource?.uuid,
  }),
};

export const InternalIpsList = connectTable(TableOptions)(TableComponent);
