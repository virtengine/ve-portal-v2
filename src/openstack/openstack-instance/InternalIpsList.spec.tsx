import { InternalIP } from '@waldur/resource/types';
import { renderTable } from '@waldur/table/testUtils';

import { InternalIpsList } from './InternalIpsList';

const renderList = () => {
  const item: InternalIP = {
    fixed_ips: [{ ip_address: '192.168.42.14', subnet_id: '' }],
    mac_address: 'fa:16:3e:93:f0:7d',
    subnet: 'api/openstacktenant-subnets/7350f289a6d14e4bbd780ee59b2899e6/',
    subnet_uuid: '7350f289a6d14e4bbd780ee59b2899e6',
    subnet_name: 'theses-and-papers-on-mach-sub-net',
    subnet_description: '',
    subnet_cidr: '192.168.42.0/24',
    allowed_address_pairs: [],
  };
  return renderTable(InternalIpsList, 'openstack-internal-ips', '0', item);
};

describe('InternalIpsList', () => {
  it('renders list', () => {
    const wrapper = renderList();
    expect(wrapper.find('tbody').html()).toMatchSnapshot();
  });
});
