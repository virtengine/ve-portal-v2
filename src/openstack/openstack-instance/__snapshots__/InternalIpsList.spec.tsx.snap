// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InternalIpsList renders list 1`] = `"<tr><td>192.168.42.14</td><td>fa:16:3e:93:f0:7d</td><td>theses-and-papers-on-mach-sub-net</td><td>192.168.42.0/24</td><td><button type=\\"button\\" class=\\"btn btn-sm btn-default\\"><i class=\\"fa fa-pencil\\"></i> Set allowed address pairs</button></td></tr>"`;
